pkg/api/v1alpha1/cloudstackmachineconfig_types.go

```diff
@@ -147,6 +148,22 @@ func (r SymlinkMaps) Validate() (err error, field string, value string) {
 	return nil, "", ""
 }
 
+// ValidateUsers verifies a CloudStackMachineConfig object must have a users with ssh authorized keys.
+// This validation only runs in CloudStackMachineConfig validation webhook, as we support
+// auto-generate and import ssh key when creating a cluster via CLI.
+func (c *CloudStackMachineConfig) ValidateUsers() error {
```

Comment:
why not have a default webhook?

Reply:
Do we need both `setCloudStackMachineConfigUserDefaults` and `ValidateUsers` to have the same logic?

Reply:
We won't support users for the web-hook because only the user name can be assigned by default and they are required to specify the `name` field anyway. If we can't default the users in full, I think it makes sense to just validate it.

It also decreases confusion on what to

Reply:
`setCloudStackMachineConfigUserDefaults` and `ValidateUsers` don't necessarily have the same logic. One mutates the `machineconfig`, while one just validates

Reply:
There might be a misunderstand. We're not using the validation webhook to perform defaulting.

The `users` are only being defaulted for the CLI because we auto generate the ssh keys. For the webhook, we opted to remove defaulting users entirely for just validations because we don't currently support auto generating ssh keys.

Reply:
Resolved in direct messages